packages/web-app-files/src/composables/actions/files/useFileActionsPaste.ts

```diff
@@ -27,17 +35,42 @@ export const useFileActionsPaste = ({ store }: { store?: Store<any> } = {}) => {
     return $pgettext('Keyboard shortcut for non-macOS systems for pasting files', 'Ctrl + V')
   })
 
-  const handler = ({ space }: FileActionOptions) => {
-    store.dispatch('Files/pasteSelectedFiles', {
-      targetSpace: space,
-      clientService,
-      loadingService,
-      createModal: (...args) => store.dispatch('createModal', ...args),
-      hideModal: (...args) => store.dispatch('hideModal', ...args),
-      showMessage: (...args) => store.dispatch('showMessage', ...args),
-      $gettext,
-      $gettextInterpolate,
-      $ngettext
+  const handler = async ({ space: targetSpace }: FileActionOptions) => {
+    const resourceSpaceMapping: Record<string, { space: SpaceResource; resources: Resource[] }> =
+      store.getters['Files/clipboardResources'].reduce((acc, resource) => {
+        const matchingSpace = getMatchingSpace(resource)
+
+        if (!(matchingSpace.id in acc)) {
+          acc[matchingSpace.id] = { space: matchingSpace, resources: [] }
+        }
+
+        acc[matchingSpace.id].resources.push(resource)
+        return acc
+      }, {})
+    const promises = Object.values(resourceSpaceMapping).map(
+      ({ space: sourceSpace, resources: resourcesToCopy }) => {
+        return loadingService.addTask(
+          () => {
+            return store.dispatch('Files/pasteSelectedFiles', {
+              targetSpace,
+              sourceSpace: sourceSpace,
+              resources: resourcesToCopy,
+              clientService,
+              loadingService,
+              createModal: (...args) => store.dispatch('createModal', ...args),
+              hideModal: (...args) => store.dispatch('hideModal', ...args),
+              showMessage: (...args) => store.dispatch('showMessage', ...args),
+              $gettext,
+              $gettextInterpolate,
+              $ngettext
+            })
+          },
+          { debounceTime: 0 }
+        )
+      }
+    )
+    return Promise.all(promises).then(() => {
```

Comment:
await

Reply:
enable conflict handling when 2 resources with same name from different spaces will be pasted